### PR TITLE
[Feature] MetaboliteSpectralMatcher optional spectra merging and export of native spectrum IDs

### PR DESCRIPTION
**Pull request by axelwalter**
## Description

In MetaboliteSpectralMatcher MS2 spectra with same precursor mass get merged by default. Result is a smaller peak map and reported spectrum indexes that don't match to the original peak map any more.
Added a parameter for optional spectrum merging and optional output of the merged peak map as mzML file.
During spectrum merging, the native IDs of merged spectra get combined in a comma separated consensus native ID instead of taking the native ID of only the first of the merged spectra.
MetaboliteSpectralMatcher now reports the native IDs of observed spectra in a new column.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Review by eeko-kon — APPROVED** [review]
Tested the changed files and it works (merging spectra should be explained at documentation- possibly most would avoid it)

**Comment by axelwalter** [comment]
Thanks for testing! Agreed, an explanation would be useful here.

File: src/tests/class_tests/openms/data/FIAMS_output/SerumTest_10.mzTab [hunk 1]
```diff
@@ -1,4 +1,4 @@
-MTD	mzTab-version	1.0.0
-MTD	mzTab-mode	null
-MTD	mzTab-type	null
-MTD	description	null
+MTD	mzTab-version	1.0.0
```
Comment (timosachsenberg) [inline src/tests/class_tests/openms/data/FIAMS_output/SerumTest_10.mzTab:1]: I think this was accidently checked in?

Comment (axelwalter) [reply src/tests/class_tests/openms/data/FIAMS_output/SerumTest_10.mzTab:1]: yes, will remove it

File: src/utils/MetaboliteSpectralMatcher.cpp [hunk 2]
```diff
@@ -103,6 +106,8 @@ class TOPPMetaboliteSpectralMatcher :
     setValidFormats_("database", {"mzML", "msp", "mgf"});
     registerOutputFile_("out", "<file>", "", "mzTab file");
     setValidFormats_("out", ListUtils::create<String>("mzTab"));
+    registerOutputFile_("out_merged_spectra", "<file>", "", "Output merged spectra as mzML file if algorithm:merge_spectra is set to true (default).", false);
```
Comment (jpfeuffer) [inline src/utils/MetaboliteSpectralMatcher.cpp:109]: What happens if the outputfile is empty but merge_spectra is true?

Keep in mind that the passing of outputfiles is not obvious in UI based workflow managers. By default, an output file is always given. If it is not created by the tool, the workflow will be in trouble.

Comment (axelwalter) [reply src/utils/MetaboliteSpectralMatcher.cpp:109]: In that case the spectra get merged but no output file is written.
The parameter `out_merged_spectra` is set as optional. Does this get handled as an optional output file like in other TOPP tools?

Comment (timosachsenberg) [reply src/utils/MetaboliteSpectralMatcher.cpp:109]: One possibility that is more consistent would be to make an optional "out_spectra"  parameter that gets written in both cases (with/without merging).

Comment (jpfeuffer) [reply src/utils/MetaboliteSpectralMatcher.cpp:109]: Yes, if wanting both is rare, this is a much better option

Comment (axelwalter) [reply src/utils/MetaboliteSpectralMatcher.cpp:109]: Right now it doesn't get written if merging is deactivated. But of course we could write in both cases as suggested. I will comment accordingly in the docs.

**Review by timosachsenberg — APPROVED** [review]

**Pull request merged**